pkg/console/controllers/service/controller.go

```diff
@@ -0,0 +1,148 @@
+package service
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/openshift/library-go/pkg/operator/management"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	// key is basically irrelevant
+	controllerWorkQueueKey = "service-sync-work-queue-key"
+	controllerName         = "ConsoleServiceSyncController"
+)
+
+// ctrl just needs the clients so it can make requests
+// the informers will automatically notify it of changes
+// and kick the sync loop
+type ServiceSyncController struct {
+	// clients
+	// consoleConfigClient  v1.ConsoleInterface
+	operatorConfigClient v1helpers.OperatorClient
+	// listers
+	serviceClient v1.ServiceLister
+	// names
+	targetNamespace string
+	serviceName     string
+	// events
+	cachesToSync []cache.InformerSynced
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+// factory func needs clients and informers
+// informers to start them up, clients to pass
+func NewServiceSyncController(
+	// clients
+	// NOTE: this is lib-go client, not pkg/console/operatorclient
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	kubeInformersNamespaced informers.SharedInformerFactory,
+	// names
+	targetNamespace string,
+	serviceName string,
+	// events
+	recorder events.Recorder,
+) *ServiceSyncController {
+
+	coreInformers := kubeInformersNamespaced.Core().V1()
+	serviceInformer := coreInformers.Services()
+	serviceLister := serviceInformer.Lister()
+
+	ctrl := &ServiceSyncController{
+		// clients
+		operatorConfigClient: operatorClient,
+		serviceClient:        serviceLister,
+		// names
+		targetNamespace: targetNamespace,
+		serviceName:     serviceName,
+		// events
+		recorder:     recorder,
+		cachesToSync: nil,
+		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
+	}
+
+	operatorClient.Informer().AddEventHandler(ctrl.newEventHandler())
+	serviceInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	ctrl.cachesToSync = append(ctrl.cachesToSync,
+		operatorClient.Informer().HasSynced,
+		serviceInformer.Informer().HasSynced,
+	)
+
+	return ctrl
+}
+
+func (c *ServiceSyncController) sync() error {
+	// spec, status, resourcesVerion, err
+	spec, _, _, err := c.operatorConfigClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+	// only run if we are managed
+	if !management.IsOperatorManaged(spec.ManagementState) {
+		return nil
+	}
+
+	// if err := SyncService(); err != nil {
```

Comment:
TODO: here is where the current `service` sync logic will be wired in once removed from the original sync loop.